exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/output/DecimalReturnTypeInference.java

```diff
@@ -277,7 +277,7 @@
 
       return TypeProtos.MajorType.newBuilder()
           .setMinorType(attributes.getReturnValue().getType().getMinorType())
-          .setScale(scale)
+          .setScale(Math.max(scale, 0))
```

Comment:
How about adjusting precision/scale for other `ReturnTypeInference` like `DecimalAggReturnTypeInference, DecimalMaxScaleReturnTypeInference, etc`